### ISSUE TITLE
Render actor icons in viewport at constant size

### ISSUE THREAD

**Issue by xxSeys1** [FlaxEngine/FlaxEngine#2644]
**Issue description:**

Actor icons for lights and stuff can become very small (even when you're not ridiculously far away from them). The capsule in the bottom right corner is aprox 2 meters tall:
![image](https://github.com/FlaxEngine/FlaxEngine/assets/111653551/20289d2a-658c-44c1-b592-4fc094c3e59a)

This can make it hard to see and select them.

Godot for example just renders every actor icon at the same size regardless of distance to the camera and I think that's basically the way to go (0.5 m radius sphere as a scale reference next to the light):
![image](https://github.com/FlaxEngine/FlaxEngine/assets/111653551/7803616a-86c0-4f2c-8615-5dba103ab9f8)

This could lead to issues of actor icons becoming too big for some ppl, so I think we should add an *Actor icons scale* slider in the view menu:
![image](https://github.com/FlaxEngine/FlaxEngine/assets/111653551/9568f5f0-a7d7-4f1a-ab8d-1ae760cba0fb)

**Steps to reproduce:**

1. Add f.e. a light into ur scene
2. Fly very far away with the camera

**Minimal reproduction project:**

none

**Flax version:**

1.8.2

**Comment by mafiesto4** [comment]
This can be implemented only as an optional feature (as suggested) because the icons would be too messy in large scenes.

**Comment by Myterian** [comment]
This could be distance based, too. Like the further you get away, the smaller the icons get until they disappear. To keep the viewport clean. And then the size is multiplied by the proposed _Icon Scale Slider_

**Comment by Myterian** [comment]
I found a temporary solution for this: You can alter the default editor sprite material found in the content window under `Flax/Content/Editor/Icons`. There you can change it from this:

![grafik](https://github.com/FlaxEngine/FlaxEngine/assets/30194679/34aeb533-b510-45fd-b8de-dce9f31c35f8)

to this:

![grafik](https://github.com/FlaxEngine/FlaxEngine/assets/30194679/08b61023-0453-4089-84ab-dc8c09599978)

and have the icons as big as you want ... until the next time you update the engine. Maybe the size could be a paramter that is set in the viewport menu and then applied to the material. @mafiesto4 I this can help you :)

On that note, maybe we could have a paramter for the delpth test too. It's a personal thing, but I prefer to see the icons thru wall and stuff.

**Comment by xxSeys1** [comment]
Improved it to scale based on distance to camera with a clamp for size
![image](https://github.com/FlaxEngine/FlaxEngine/assets/111653551/c0595403-7787-489d-8453-278d3d84f3ea)

It's a material function, just drop it into `Flax/Content/Editor/Icons/IconsMaterial` and connect to *Position Offset*.

Only two issues are that this increases base icon size and does not increase the area to select the associated actor. The first one can prolly be fixed relatively easy but I haven't figured it out yet.

[MF_IconFix.zip](https://github.com/FlaxEngine/FlaxEngine/files/15448050/MF_IconFix.zip)

**Comment by mafiesto4** [comment]
That's cool but it won't work with selection as users need to be able to select those icons too. Need to add some code there too.

**Comment by xxSeys1** [comment]
> That's cool but it won't work with selection as users need to be able to select those icons too. Need to add some code there too.

Yup. I think I will make a pr addressing this issue some time in the near future. Or are you working on something like this right now?

**Comment by mafiesto4** [comment]
I'm gonna add it quickly

**Comment by mafiesto4** [comment]
Added in f2aaad004830e91ce8a075b774de708529a28228
